recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -330,19 +432,26 @@ def main(argv):
     parser.add_argument('-f', '--force',
                         help='Force update, ignore all the checking',
                         action='store_true')
+    parser.add_argument('-p', '--preserve_list',
+                        type=str,
+                        help='env preserve list')
 
     try:
         args = parser.parse_args()
     except IOError as e:
         print(e.strerror, file=sys.stderr)
         exit(1)
 
-    erase_env_input = input("\nWarning: All U-Boot environment variables will be reset to factory settings. Continue (y/N)? ")
+    erase_env_input = input(
+        "\nWarning: All U-Boot environment variables will be reset to factory settings.\n"
+        "Unless preserve env in the update.conf.json or using -p argument.Continue(y/N)?")
```

Comment:
This could be made more concrete: You do not need to warn if there are no other user modifications noticed and `-p` is provided.